### ISSUE TITLE
feat: implement CanvasScreen — WebView UI, auth headers, and CanvasController wiring

### ISSUE THREAD

**Issue by yuga-hashimoto** [yuga-hashimoto/OpenClawAssistant#223]
## Overview

`CanvasController.kt` (functional layer) is fully implemented — `canvas.navigate`, `canvas.eval`, `canvas.snapshot`, A2UI push/reset are all working. What's missing is the **UI layer**: the `CanvasScreen.kt` composable that hosts the WebView, wires it to `CanvasController`, and injects Gateway auth headers.

## Current State

| Component | Status |
|---|---|
| `node/CanvasController.kt` (navigate, eval, snapshot, A2UI) | ✅ Complete |
| `node/InvokeDispatcher.kt` (canvas.* routing) | ✅ Complete |
| `ui/CanvasScreen.kt` composable | ❌ Missing |
| `CanvasController.detach()` | ❌ Missing |
| Gateway auth headers in WebView requests | ❌ Missing |
| Tab navigation entry point to CanvasScreen | ❌ Missing (see #203) |

---

## Task 1: `CanvasScreen.kt` composable

**File:** `app/src/main/java/com/openclaw/assistant/ui/CanvasScreen.kt`

```kotlin
@Composable
fun CanvasScreen(
    canvasController: CanvasController,
    modifier: Modifier = Modifier
) {
    AndroidView(
        modifier = modifier.fillMaxSize(),
        factory = { context ->
            WebView(context).also { webView ->
                webView.settings.apply {
                    javaScriptEnabled = true
                    domStorageEnabled = true
                    allowFileAccess = true
                    mixedContentMode = WebSettings.MIXED_CONTENT_COMPATIBILITY_MODE
                }
                webView.webViewClient = CanvasWebViewClient(canvasController)
                canvasController.attach(webView)
            }
        },
        onRelease = { webView ->
            webView.stopLoading()
            canvasController.detach()
            webView.destroy()
        }
    )
}
```

---

## Task 2: `CanvasWebViewClient`

Handles `onPageFinished` (debug status overlay) and Gateway auth header injection for resource requests.

```kotlin
class CanvasWebViewClient(
    private val canvasController: CanvasController
) : WebViewClient() {

    override fun onPageFinished(view: WebView, url: String) {
        super.onPageFinished(view, url)
        canvasController.onPageFinished()
    }

    override fun shouldInterceptRequest(
        view: WebView,
        request: WebResourceRequest
    ): WebResourceResponse? {
        val token = canvasController.gatewayToken?.takeIf { it.isNotBlank() } ?: return null
        val origin = canvasController.gatewayOrigin ?: return null
        if (!request.url.toString().startsWith(origin)) return null

        return try {
            val conn = URL(request.url.toString()).openConnection() as HttpURLConnection
            request.requestHeaders.forEach { (k, v) -> conn.setRequestProperty(k, v) }
            conn.setRequestProperty("Authorization", "Bearer $token")
            conn.connect()
            WebResourceResponse(
                conn.contentType?.substringBefore(";") ?: "text/html",
                conn.contentEncoding ?: "utf-8",
                conn.inputStream
            )
        } catch (_: Exception) { null }
    }
}
```

---

## Task 3: Gateway auth in `CanvasController`

Add token/origin fields and pass headers to the initial `loadUrl` call.

```kotlin
class CanvasController {
    // Add:
    @Volatile var gatewayToken: String? = null
    @Volatile var gatewayOrigin: String? = null

    fun setGatewayAuth(origin: String?, token: String?) {
        gatewayOrigin = origin
        gatewayToken = token
    }

    private fun authHeaders(): Map<String, String> {
        val t = gatewayToken?.takeIf { it.isNotBlank() } ?: return emptyMap()
        return mapOf("Authorization" to "Bearer $t")
    }

    private fun reload() {
        withWebViewOnMain { wv ->
            if (url == null) {
                wv.loadUrl(scaffoldAssetUrl)          // local asset — no auth needed
            } else {
                wv.loadUrl(url!!, authHeaders())      // gateway URL — send Bearer token
            }
        }
    }
}
```

Wire up in `NodeRuntime` / `ConnectionManager` after successful gateway auth:

```kotlin
canvasController.setGatewayAuth(
    origin = endpoint.baseUrl,   // e.g. "http://192.168.1.10:4000"
    token  = authToken
)
```

---

## Task 4: `CanvasController.detach()`

Currently only `attach(webView)` exists. Add the matching cleanup method:

```kotlin
fun detach() {
    webView = null
}
```

---

## Files to Change / Create

| File | Change |
|---|---|
| `ui/CanvasScreen.kt` *(new)* | `CanvasScreen` composable + `CanvasWebViewClient` |
| `node/CanvasController.kt` | Add `gatewayToken`, `gatewayOrigin`, `setGatewayAuth()`, `detach()`, pass headers to `loadUrl` |
| `node/NodeRuntime.kt` or `gateway/ConnectionManager.kt` | Call `canvasController.setGatewayAuth(...)` after auth success, clear on disconnect |

Navigation to CanvasScreen from the main UI is tracked separately in **#203**.

---

## Acceptance Criteria

- [ ] `CanvasScreen` composable exists and displays a WebView filling the screen
- [ ] WebView settings: JS enabled, DOM storage enabled, file access allowed, mixed content compat mode
- [ ] `CanvasController.attach(webView)` called on WebView creation; `detach()` + `destroy()` called on disposal
- [ ] `canvasController.onPageFinished()` called from `WebViewClient.onPageFinished()`
- [ ] `canvas.navigate` to a Gateway URL includes `Authorization: Bearer <token>` on the initial `loadUrl`
- [ ] In-page resource requests to the same gateway origin also include the auth header via `shouldInterceptRequest`
- [ ] No auth header is added when token is null/blank (unauthenticated setups unaffected)
- [ ] Local scaffold asset (`file:///android_asset/...`) is never modified

## Dependency

- **#203** — Compose Navigation: needed to make CanvasScreen reachable from the main UI
